ihmc-high-level-behaviors/src/main/java/us/ihmc/perception/RapidHeightMapManager.java

```diff
@@ -48,11 +54,19 @@ public RapidHeightMapManager(OpenCLManager openCLManager,
                                 CameraIntrinsics depthImageIntrinsics,
                                 ROS2PublishSubscribeAPI ros2)
    {
-      rapidHeightMapExtractor = new RapidHeightMapExtractorCUDA(leftFootSoleFrame, rightFootSoleFrame);
-      rapidHeightMapExtractor.setDepthIntrinsics(depthImageIntrinsics);
+      if (hasCUDAAvailable)
+      {
+         deviceDepthImage = new GpuMat(depthImageIntrinsics.getWidth(), depthImageIntrinsics.getHeight(), opencv_core.CV_16UC1);
+         rapidHeightMapExtractor = new RapidHeightMapExtractorCUDA(leftFootSoleFrame, rightFootSoleFrame, deviceDepthImage, 1);
```

Comment:
If the CUDA height map has bugs like you said, why are we using it in RapidHeightMapManager at the moment? It's not clear what this PR is trying to do. I would expect that the CUDA implementation only be in some demo/development classes until it's ready to be swapped out.

Reply:
This PR supports using CUDA, when its available. And if its not available, we use the OpenCL height map.

Reply:
I don't think the RapidHeightMapManager should support both. I think it should support the known working implementation (OpenCL), and when the CUDA height map extractor is ready, we switch it out.